src/components/application_manager/test/policy_handler_test.cc

```diff
@@ -1467,24 +1668,919 @@ TEST_F(PolicyHandlerTest, SendMessageToSDK) {
   EnablePolicyAndPolicyManagerMock();
   test_app.insert(mock_app_);
   // Expectations
-  EXPECT_CALL(app_manager_, application(kAppId_))
+  EXPECT_CALL(app_manager_, application(kAppId1_))
       .WillRepeatedly(Return(mock_app_));
   EXPECT_CALL(*mock_app_, policy_app_id()).WillOnce(Return(kPolicyAppId_));
 
-  EXPECT_CALL(*MockMessageHelper::message_helper_mock(),
-              SendPolicySnapshotNotification(kAppId_, msg, url, _));
   // Act
-  policy_handler_.last_used_app_ids().push_back(kAppId_);
-  EXPECT_CALL(*MockMessageHelper::message_helper_mock(),
-              SendPolicySnapshotNotification(kAppId_, msg, url, _));
+  policy_handler_.last_used_app_ids().push_back(kAppId1_);
+  EXPECT_CALL(mock_message_helper_,
+              SendPolicySnapshotNotification(kAppId1_, msg, url, _));
+  EXPECT_TRUE(policy_handler_.SendMessageToSDK(msg, url));
+}
+
+TEST_F(PolicyHandlerTest, SendMessageToSDK_InavalidApp_UNSUCCESS) {
+  BinaryMessage msg;
+  const std::string url = "test_url";
+  EnablePolicyAndPolicyManagerMock();
+  utils::SharedPtr<application_manager_test::MockApplication> invalid_app;
+  policy_handler_.last_used_app_ids().push_back(kAppId1_);
+
+  EXPECT_CALL(app_manager_, application(kAppId1_))
+      .WillOnce(Return(invalid_app));
+  EXPECT_CALL(*mock_app_, policy_app_id()).Times(0);
+
+  EXPECT_FALSE(policy_handler_.SendMessageToSDK(msg, url));
+}
+
+#ifdef SDL_REMOTE_CONTROL
+TEST_F(PolicyHandlerTest,
+       SendMessageToSDK_RemoteControlInvalidMobileAppId_UNSUCCESS) {
+  // Precondition
+  BinaryMessage msg;
+  const std::string url = "test_url";
+  EnablePolicyAndPolicyManagerMock();
+  EXPECT_CALL(app_manager_, active_application()).WillOnce(Return(mock_app_));
+  EXPECT_CALL(*mock_app_, app_id()).WillOnce(Return(kAppId1_));
+  EXPECT_CALL(app_manager_, application(kAppId1_))
+      .WillRepeatedly(Return(mock_app_));
+  const std::string empty_mobile_app_id("");
+  EXPECT_CALL(*mock_app_, policy_app_id())
+      .WillOnce(Return(empty_mobile_app_id));
+
+  EXPECT_CALL(mock_message_helper_, SendPolicySnapshotNotification(_, _, _, _))
+      .Times(0);
+  EXPECT_FALSE(policy_handler_.SendMessageToSDK(msg, url));
+}
+
+TEST_F(PolicyHandlerTest, SendMessageToSDK_RemoteControl_SUCCESS) {
+  BinaryMessage msg;
+  const std::string url = "test_url";
```

Comment:
@okozlovlux, You use this kind of variable in several test cases. May you move it to an nameless namespace or fixture class?